python/huggingfaceserver/huggingfaceserver/model.py

```diff
@@ -108,31 +109,46 @@ def load(self) -> bool:
                 self.vllm_engine = AsyncLLMEngine.from_engine_args(self.vllm_engine_args)
                 return True
 
+        model_config = AutoConfig.from_pretrained(model_id_or_path)
+
         if not self.task:
-            self.task = self.infer_task_from_model_architecture(model_id_or_path)
+            self.task = self.infer_task_from_model_architecture(model_config)
+
+        # device_map = "auto" enables model parallelism but all model architcture dont support it.
+        # For pre-check we initialize the model class without weights to check the `_no_split_modules`
+        # device_map = "auto" for models that support this else set to either cuda/cpu
+        with init_empty_weights():
+            self.model = AutoModel.from_config(model_config)
+
+        if self.model._no_split_modules:
+            self.device_map = "auto"
```

Comment:
should we call `infer_auto_device_map` here?

Reply:
https://huggingface.co/docs/accelerate/en/concept_guides/big_model_inference

Reply:
I think the functionality wise it's the same